### ISSUE TITLE
Redirecting to intended URL instead of always to the homepage

### ISSUE THREAD

**Issue by laurensgroeneveld** [auth0/laravel-auth0#272]
### What should be added?

In the current version of this package, the user is always redirected to (a configurable) home route upon successful login. However, I find that that leads to a sometimes jarring experience. For example when the user is not logged in and clicks a link to the application in an email, they end up on the homepage rather than the intended URL. As far as I could find, there is currently no way to configure this.

My current solution for this is to copy the Login and Callback controllers into my application code (as they work fine for my usecase otherwise), and updating their redirects by basically putting the intended URL in the session, and defaulting to a route if it's not there.

```php
// Login
session()->put(['auth_return_to_url' => url()->previous()]);

// Callback
$redirectTo = session()->pull('auth_return_to_url', route('members'));

return redirect()->intended($redirectTo);
```

Is this functionality something that could be part of this package? If so, I could take a stab at implementing this in a PR.

**Comment by evansims** [comment]
Hey @laurensgroeneveld! 👋 Thanks for your feedback and interest.

To me, this sounds like a perfect use case for leveraging the [AuthenticationSucceeded event](https://github.com/auth0/laravel-auth0/blob/main/src/Event/Stateful/AuthenticationSucceeded.php), which [is fired](https://github.com/auth0/laravel-auth0/blob/main/src/Http/Controller/Stateful/Callback.php#L66-L67) from the callback route controller after a user authenticates successfully.

You could store any state information you want (like a return URL) before sending the user through the authentication flow, and using [an event listener](https://laravel.com/docs/9.x/events#defining-listeners) within your application, listen for and react upon receiving the AuthenticationSucceeded event. You could do anything you need from there, including redirect them using the previous defined state data you saved.

I think this is much more flexible and offer far greater customization potential than anything we could provide a configuration interface around ourselves. Would this approach address your concern?

**Comment by laurensgroeneveld** [comment]
Thanks for your quick answer @evansims!

> I think this is much more flexible and offer far greater customization potential than anything we could provide a configuration interface around ourselves. Would this approach address your concern?

Yes, makes sense! Redirecting to the previous page feels like a common enough use case but this way is still fairly simple to implement without a lot of boilerplate, so it's fine with me 👍 

I was in the process of upgrading from v6 to v7 and hadn't noticed these events where there, maybe they can be documented in the readme?

**Comment by evansims** [comment]
Thanks @laurensgroeneveld, I appreciate the feedback! I agree the README at the very least should be updated to cover this topic better, that's on me; I'll get that updated this week with more info on the event system. Let me know if you encounter any problems or have any other questions/feedback on the library!

**Comment by laurensgroeneveld** [comment]
After a bit of experimenting I ended up with a slightly different approach, by implementing my own authentication middleware. It turns out the Laravel redirector already has a [setIntendedUrl](https://github.com/laravel/framework/blob/9.x/src/Illuminate/Routing/Redirector.php#L120) method that sets a cookie which is [used](https://github.com/laravel/framework/blob/9.x/src/Illuminate/Routing/Redirector.php#L107) in the `intended` method. So I adapted the middleware present in this package by adding a call to `setIntendedUrl`:

```php
redirect()->setIntendedUrl($request->url());
```

I added a little bit of additional logic around this to ensure this only happens when the url of the request is actually for a page that is protected by login.

This approach feels pretty lightweight. Note that I don't use `redirect()->intended(...)` anywhere else in the app, otherwise this could have unforeseen side effects as that intended url is set on every request to a route that has this middleware.
